### DISCUSSION TITLE
Support for Python-like languages?

### DISCUSSION THREAD

**Discussion by VinnyVicious** [pest-parser/pest#564] [Q&A]
Is Pest able to parse grammars like Python, where indentation is important?

**Comment by CAD97** [comment]
Pest doesn't have any automatic whitespace handling, so it's fully possible.

The `WHITESPACE` magic rule allows `~` to skip whatever it matches, typically being trivia such as whitespace and comments.

For actually handling the indentation state, you can use atomic rules (which don't implicitly eat whitespace) and `PUSH`/`PEEK_ALL`/`POP` to maintain an indentation stack and maintain it. Unfortunately, actually doing so is far from trivial.
